processor/src/host/debug.rs

```diff
@@ -71,6 +74,44 @@ impl Printer {
         }
     }
 
+    /// Prints the number of advice stack items specified by `n` if it is provided, otherwise print
+    /// only the top.
+    fn print_vm_adv_stack(
+        &self,
+        advice_provider: &impl AdviceProvider,
+        process: ProcessState,
+        requested: Option<usize>,
```

Comment:
*Nit:* I'd either make this `Option<NonZeroUsize>` or interpret `Some(0)` the same as `None`, i.e. no limit to the number of stack elements printed/print the entire stack.